### PR TITLE
Add next pages examples for chat

### PR DESCRIPTION
**Pull request by jeremyphilemon**

### CODE REVIEW

File: content/examples/02-next-pages/05-chat/01-generate-chat-completion.mdx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        onKeyDown={async event => {
+          if (event.key === 'Enter') {
+            setMessages([...messages, { role: 'user', content: input }]);
+
+            await fetch('/api/chat', {
+              method: 'POST',
+              body: JSON.stringify({
+                messages: [...messages, { role: 'user', content: input }],
+              }),
+            }).then(async response => {
```
Comment (lgrammel) [inline content/examples/02-next-pages/05-chat/01-generate-chat-completion.mdx:54]: combining `await` and `then` seems off. Can you change to async/await here? `const response = await fetch ...`

File: content/examples/02-next-pages/05-chat/01-generate-chat-completion.mdx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+export async function POST(req: Request) {
+  const { messages }: { messages: CoreMessage[] } = await req.json();
+
+  const { responseMessages } = await generateText({
+    model: openai('gpt-4'),
+    system: 'You are a helpful assistant.',
+    messages,
+  });
+
+  return Response.json({ messages: responseMessages });
```
Comment (lgrammel) [inline content/examples/02-next-pages/05-chat/01-generate-chat-completion.mdx:100]: nice, didn't think about using `responseMessages` like this

File: content/examples/02-next-pages/05-chat/02-stream-chat-completion.mdx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+export async function POST(req: Request) {
+  const { messages }: { messages: CoreMessage[] } = await req.json();
+
+  const { responseMessages } = await generateText({
+    model: openai('gpt-4'),
+    system: 'You are a helpful assistant.',
+    messages,
+  });
+
+  return Response.json({ messages: responseMessages });
```
Comment (lgrammel) [inline content/examples/02-next-pages/05-chat/02-stream-chat-completion.mdx:74]: Does this work with `useChat`? i feel this example might be broken

File: content/examples/02-next-pages/05-chat/01-generate-chat-completion.mdx (hunk 1 above)
Comment (jeremyphilemon) [reply content/examples/02-next-pages/05-chat/01-generate-chat-completion.mdx]: done!

**Review by lgrammel — APPROVED** [review]

**Pull request merged**